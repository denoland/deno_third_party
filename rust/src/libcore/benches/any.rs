// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
// file at the top-level directory of this distribution and at
// http://rust-lang.org/COPYRIGHT.
//
// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
// option. This file may not be copied, modified, or distributed
// except according to those terms.

use core::any::*;
use test::{Bencher, black_box};

#[bench]
fn bench_downcast_ref(b: &mut Bencher) {
    b.iter(|| {
        let mut x = 0;
        let mut y = &mut x as &mut Any;
        black_box(&mut y);
        black_box(y.downcast_ref::<isize>() == Some(&0));
    });
}
