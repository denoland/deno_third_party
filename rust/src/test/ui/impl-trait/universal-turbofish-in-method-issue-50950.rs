// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
// file at the top-level directory of this distribution and at
// http://rust-lang.org/COPYRIGHT.
//
// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
// option. This file may not be copied, modified, or distributed
// except according to those terms.

use std::any::Any;
pub struct EventHandler {
}

impl EventHandler
{
    pub fn handle_event<T: Any>(&mut self, _efunc: impl FnMut(T)) {}
}

struct TestEvent(i32);

fn main() {
    let mut evt = EventHandler {};
    evt.handle_event::<TestEvent, fn(TestEvent)>(|_evt| {
        //~^ ERROR cannot provide explicit type parameters
    });
}
