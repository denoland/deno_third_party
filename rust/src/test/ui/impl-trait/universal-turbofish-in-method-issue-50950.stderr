error[E0632]: cannot provide explicit type parameters when `impl Trait` is used in argument position.
  --> $DIR/universal-turbofish-in-method-issue-50950.rs:14:9
   |
LL |     evt.handle_event::<TestEvent, fn(TestEvent)>(|_evt| {
   |         ^^^^^^^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0632`.
